pip/flatpak-pip-generator

```diff
@@ -251,19 +255,29 @@ with tempfile.TemporaryDirectory(prefix=tempdir_prefix) as tempdir:
         except FileNotFoundError:
             pass
 
-    fprint('Downloading arch independent packages')
-    for filename in os.listdir(tempdir):
-        if not filename.endswith(('bz2', 'any.whl', 'gz', 'xz', 'zip')):
-            version = get_file_version(filename)
-            name = get_package_name(filename)
-            url = get_tar_package_url_pypi(name, version)
-            print('Deleting', filename)
-            try:
-                os.remove(os.path.join(tempdir, filename))
-            except FileNotFoundError:
-                pass
-            print('Downloading {}'.format(url))
-            download_tar_pypi(url, tempdir)
+    if opts.arch_dependent_force:
+        fprint('Not using arch independent packages [--arch-dependent-force]')
+    else:
+        fprint('Downloading arch independent packages')
+        for filename in os.listdir(tempdir):
+            if not filename.endswith(('bz2', 'any.whl', 'gz', 'xz', 'zip')):
+                version = get_file_version(filename)
+                name = get_package_name(filename)
+                try:
+                    url = get_tar_package_url_pypi(name, version)
+                except Exception as e:
+                    if not opts.arch_dependent_allow:
+                        raise
+                    else:
+                        print('Ignoring "{}" [--arch-dependent-allow]'.format(e))
+                        continue
```

Comment:
Again if I'm reading this right, no code gets executed downloading arch dependent packages after that. The rest of the loop is just skipped with `continue`, including `download_tar_pypi(url, tempdir)`. Don't suppose this is expected behavior either